cds_ils/migrator/ill/api.py

```diff
@@ -0,0 +1,202 @@
+
+
+CDS statuses  -  ILS BorrowingRequest statuses
+
+new           ?  ids: 55952,55463,55245,55128: resolve all new before migration
+cancelled     -> CANCELLED - migrated
+returned      -> RETURNED - migrated
+on loan       -> ON_LOAN - migrated - 35 results, create loan?
+received      -> REQUESTED - until library advices otherwise
+requested     -> REQUESTED - migrated - 8 results
+"""
+
+import json
+
+import click
+from elasticsearch_dsl import Q
+from flask import current_app
+from invenio_app_ils.ill.api import Library
+from invenio_app_ils.ill.search import LibrarySearch
+from invenio_db import db
+
+from cds_ils.migrator.api import (
+    bulk_index_records,
+    import_record,
+    model_provider_by_rectype,
+)
+from cds_ils.migrator.errors import BorrowingRequestError, ItemMigrationError
+from cds_ils.migrator.items.api import get_item_by_barcode
+from cds_ils.migrator.patrons.api import get_user_by_legacy_id
+from cds_ils.migrator.utils import get_acq_ill_notes, get_cost
+
+
+# NOTE: There are several ill borrowing requests ~9000 without barcode.
+# we use a placeholder document for the migraiton of these items.
+MIGRATION_DOCUMENT_PID = ""
+
+
+def get_status(record):
+    """Map CDS status to a valid ILS status."""
+
+    ILS_STATUSES = [
+        "CANCELLED",
+        "ON_LOAN",
+        "PENDING",
+        "REQUESTED",
+        "RETURNED",
+    ]
+    status = record["status"]
+    if status == "cancelled":
+        return ILS_STATUSES[0]
+    if status == "on loan":
+        return ILS_STATUSES[1]
+    if status == "received":
+        return ILS_STATUSES[3]
+    if status == "requested":
+        return ILS_STATUSES[3]
+    if status == "returned":
+        return ILS_STATUSES[4]
+    raise BorrowingRequestError(
+        "This status is ambiguous for Borrowing Request!\n{}".format(record)
+    )
+
+
+def get_library_by_legacy_id(legacy_id):
+    """Search for library by legacy id."""
+    search = LibrarySearch().query(
+        "bool", filter=[Q("term", legacy_id=legacy_id)]
+    )
+    result = search.execute()
+    hits_total = result.hits.total.value
+    if not result.hits or hits_total < 1:
+        raise BorrowingRequestError(
+            "no library found with legacy id {}".format(legacy_id)
+        )
+    elif hits_total > 1:
+        raise BorrowingRequestError(
+            "found more than one library with legacy id {}".format(legacy_id)
+        )
+    else:
+        return Library.get_record_by_pid(result.hits[0].pid)
+
+
+def migrate_to_ils(record):
+    """Create a record for ILS."""
+    document_pid = None
+    try:
+        item = get_item_by_barcode(record["barcode"])
+        document_pid = item.get("document_pid")
+    except ItemMigrationError:
+        document_pid = MIGRATION_DOCUMENT_PID
+
+    # library_pid
+    library = get_library_by_legacy_id(record["id_crcLIBRARY"])
+    library_pid = library.pid.pid_value
+
+    # patron_pid
+    anonym = current_app.config["ILS_PATRON_ANONYMOUS_CLASS"](-2)
```

Comment:
code duplication with the function `get_patron_pid` in acquisition

Reply:
Nice catch, I moved it to the general utils